scripts/__tests__/svg_test.js

```diff
@@ -1,39 +1,37 @@
 const path   = require('path');
 const fs     = require('fs');
+const util   = require('util');
 const expect = require('chai').expect;
+const exec   = util.promisify(require('child_process').exec);
 const common = require('../common');
 
 const ignored_files = [
     'src/images/pages/regulation/map.svg',
 ];
 
+let changed_files = [];
+
 describe('check svg file format', () => {
-    it('should be valid svgs', () => {
-        function fetchSvgs(dir) {
-            const directory = fs.readdirSync(dir);
-            return directory.map(file => {
-                if (fs.statSync(path.join(dir, file)).isDirectory()) {
-                    return fetchSvgs(path.join(dir, file));
-                }
-                return file.match(/.*\.(svg)$/ig) ? path.join(dir, file) : false;
-            });
+    const fetchFiles = async (command) => {
+        const { stdout, stderr } = await exec(command);
+        if (stderr) {
+            throw new Error(stderr);
         }
 
-        const flatten = arr => {
-            const index = arr.findIndex(el => Array.isArray(el));
-            if (index > -1) {
-                arr.splice(index, 1, ...arr[index]);
-                return flatten(arr);
-            }
+        return stdout.split('\n').filter(dir => dir.length);
+    };
 
-            return arr;
-        };
-
-        const svgs = fetchSvgs(path.resolve(common.root_path, 'src/images'));
+    it('should be valid svgs', async () => {
+        try {
+            changed_files = [
+                ...await fetchFiles('git diff --name-only -- *.svg'),
+                ...await fetchFiles('git diff HEAD origin/master --name-only -- *.svg'),
```

Comment:
does this mean you constantly need to make sure that origin/master is up to date with upstream/master? mine never is, I can't imagine everyone else remembers to always update theirs either

Reply:
No, you don't need to keep your branch updated. So, we are looking for changed files, if nothing is changed since x commits ago, then we are good to go 👍.

Reply:
In a situation that origin had a file, but the dev's branch did not include that one, an error popped up. the new commit fixes that.